servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/TimeoutHttpRequesterFilter.java

```diff
@@ -100,4 +167,59 @@ public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strat
         // No influence since we do not block.
         return strategy;
     }
+
+    @FunctionalInterface
+    public interface TimeoutFromRequest extends Function<HttpRequestMetaData, Duration> {
+
+        /**
+         * Determine timeout duration, if present, from a request and/or apply default timeout durations.
+         *
+         * @param request the current request
+         * @return The timeout or null for no timeout
+         */
+        @Override
+        @Nullable
+        Duration apply(HttpRequestMetaData request);
+    }
+
+    /**
+     * Returns a function which uses {@link #useContextDeadlineOrDefault(Duration) useContextDeadlineOrDefault()} with
+     * the provided default duration to determine the timeout duration to be used for a provided request.
+     *
+     * @param duration default timeout duration or null for no timeout
+     * @return a function to produce a timeout based on the context deadline or specified default
+     */
+    public static TimeoutFromRequest useDefaultTimeout(@Nullable Duration duration) {
```

Comment:
If we remove `useContextDeadlineOrDefault` based on discussion in https://github.com/apple/servicetalk/pull/1456/files/19674baef2572fc09a1dfc25c25b22b34ccc9cab..f4365832f371db9c976fba790b24b1b86c5b8bd6#r602650058, consider removing this method as well.